Language/Haskell/GhcMod/Monad/Types.hs

```diff
@@ -129,10 +138,11 @@ data GhcModState = GhcModState {
       gmGhcSession   :: !(Maybe GmGhcSession)
     , gmComponents   :: !(Map ChComponentName (GmComponent GMCResolved (Set ModulePath)))
     , gmCompilerMode :: !CompilerMode
+    , gmMMappedFiles :: Map TargetId (StringBuffer, UTCTime)
```

Comment:
Scratch that about having both in `gmMMappedFIles`, that's not very feasible. But I still maintain that matching on canonical filenames is more robust than matching on module names, since latter is not always well-defined.